chipsec/helper/basehelper.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/python
+#CHIPSEC: Platform Security Assessment Framework
+#Copyright (c) 2019, Intel Corporation
+#
+#This program is free software; you can redistribute it and/or
+#modify it under the terms of the GNU General Public License
+#as published by the Free Software Foundation; Version 2.
+#
+#This program is distributed in the hope that it will be useful,
+#but WITHOUT ANY WARRANTY; without even the implied warranty of
+#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#GNU General Public License for more details.
+#
+#You should have received a copy of the GNU General Public License
+#along with this program; if not, write to the Free Software
+#Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#Contact information:
+#chipsec@intel.com
+#
+
+# Base class for the helpers
+class Helper(object):
+    class __metaclass__(type):
+        def __init__(cls, name, bases, attrs):
+            if not hasattr(cls, 'registry'):
+                cls.registry = []
+            else:
+                cls.registry.append((name, cls))
+
+    def __init__(self):
+        self.driver_loaded = False
+        self.os_system = "basehelper"
+        self.os_release = "0.0"
+        self.os_version = "0.0"
+        self.os_machine = "base"
+
+    def use_native_api(self):
+        return (not self.driver_loaded)
+
+    def create(self, start_driver):
+        if logger().VERBOSE:
+            logger().log("[helper] File Helper created")
+        return True
+
+    def start(self, start_driver, from_file=None):
+        if logger().VERBOSE:
+            logger().log("[helper] File Helper started/loaded")
+        return True
+
+    def stop( self, start_driver ):
+        if logger().VERBOSE:
+            logger().log("[helper] File Helper stopped/unloaded")
+        return True
+
+    def delete(self, start_driver):
+        if logger().VERBOSE:
+            logger().log("[helper] File Helper deleted")
+        return True

```

Comment:
Create, start, stop and delete have File Helper message and return True.  I think these should also raise exceptions so that derived classes MUST implement these functions.

Reply:
Changes are complete.  I left in the logging, do you want that gone too?

Reply:
Logging is fine but it should not reference the File Helper.  You could have it say that the method was not implemented.

Reply:
Thanks!